src/index.ts

```diff
@@ -6,41 +6,76 @@
  * MIT Licensed
  */
 
-export class Crc32 {
-  #CRC32_POLYNOMIAL = 0xedb88320;
-  #table = new Uint32Array(256).fill(0);
+class InvalidInputError extends Error {
+  constructor(public data: unknown) {
+    super("Invalid input: data must be a Uint8Array.");
+    this.name = "InvalidInputError";
+  }
+}
 
-  constructor() {
-    this.#generateTable();
+class EmptyDataArrayError extends Error {
+  constructor(public data: unknown) {
+    super("Cannot calculate CRC for an empty data array.");
```

Comment:
In this case we already know that `data` is an array so we no longer need the `unknown`. It is generally helpful if we use the most specific type when possibe. It is also a bit probelmatic semantically since the message implies that we know it is an array yet the type is `unknown` which could be puzzling for someone who reads the code.